tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/entry/function/ArgumentParserPositiveTest.java

```diff
@@ -43,12 +43,31 @@
     private ByteArrayOutputStream tempOutStream = new ByteArrayOutputStream();
     private PrintStream defaultOut;
     private ProgramFile programFile;
+    private ProgramFile mainWithParamsProgramFile;
 
     @BeforeClass
     public void setUp() {
         defaultOut = System.out;
         programFile = LauncherUtils.compile(Paths.get("src/test/resources/test-src/entry.function"),
                                                         Paths.get(FILE_NAME), false);
+        mainWithParamsProgramFile = LauncherUtils.compile(Paths.get("src/test/resources/test-src/entry.function"),
+                                                          Paths.get("test_main_with_params.bal"),
+                                                          false);
+    }
+
+    @Test (dataProvider = "mainFunctionParams")
+    public void testMainWithParams(String intString, String floatString, String expectedString) {
+        BValue[] entryFuncResult = BLangProgramRunner.runEntryFunc(mainWithParamsProgramFile, "main",
+                                                                   new String[]{intString, floatString});
```

Comment:
The argument array is only used here and extracting it as a separate variable would only give it an identifier, wouldn't make much of a difference since this is clear, right?

Reply:
Yes, that is true. But the code will look better without the leading empty spaces. Maybe that is just my personal preference 😄